spec/models/encryptable_spec.rb

```diff
@@ -84,6 +82,36 @@
     end
   end
 
+  context 'papertrail', versioning: true do
+    context 'delete' do
+      it 'deletes log history if encryptable is delted' do
+        1000.times do
+          encryptable.touch
+        end
+
+        expect(encryptable.versions.size).to eq(1000)
+
+        encryptable.destroy
+
+        expect(PaperTrail::Version.all.size).to be(0)
+      end
+    end
+
+    context 'touch' do
+      it 'creates a log entry' do
+        encryptable.touch
+        expect(encryptable.versions.count).to eq(1)
+      end
+
+      it 'contains user in whodunnit' do
+        PaperTrail.request.whodunnit = 'Bob'
+
+        encryptable.touch
+        expect(encryptable.versions.last.whodunnit).to eq('Bob')
```

Comment:
hmm ... further up `whodunit` included a foreign key (User)
now it includes the username as string.
shouldn't this be always a foreign_key?
please check how this should be done correctly

Reply:
For testing purposes, `whodunnit` is set to a `string` in the `PaperTrail` documentation.

Reply:
what are we testing here?
so far, read access seems to be logged inside the encryptables controller.
I think we can trust paper trail to set whodunnit correctly when request.whodunnit is set.
maybe this should be rather be tested in encryptables controller spec #show action